src/networking.c

```diff
@@ -3810,6 +3888,7 @@ void clientCommand(client *c) {
         /* If this client has to be closed, flag it as CLOSE_AFTER_REPLY
          * only after we queued the reply to its output buffers. */
         if (close_this_client) c->flag.close_after_reply = 1;
+        zfree(client_filter.ids);
```

Comment:
I see this zfree on error in multiple places. We should refactor it so we only need it in one place.

One way to do this is to use a goto label like `client_kill_done` just before this last zfree in the end and on error we use goto instead of zfree + return.

```suggestion
client_kill_done:
        zfree(client_filter.ids);
```

We can also consider splitting it into smaller functions, if it helps. This function is very large.

Reply:
> We can also consider splitting it into smaller functions, if it helps. This function is very large.

If its okay, I can take it up in a separate PR? Both PRs would be easier to review separately.

Reply:
No, let's not add technical dept with some plan to fix it later. Many times, that kind of follow-up never happens. Review is not too hard.

If you want, you can do the refactoring in separate commits within the same PR. It's possible to look at the diff commit-per-commit.